plugins/worker_api/worker_api_plugin.cpp

```diff
@@ -76,11 +89,26 @@ void worker_api_plugin::worker_api_plugin_impl::select_postbased_results_ordered
 
         result.reserve(query.limit);
 
-        for (; itr != idx.end() && result.size() < query.limit; ++itr) {
+        auto handle = [&]() {
             if (!select(query, *itr)) {
-                continue;
+                return;
+            }
+            if (fill_posts) {
+                const auto& comment = _db.get_comment(itr->author, itr->permlink);
+                result.emplace_back(*itr, helper->create_comment_api_object(comment));
+            } else {
+                result.emplace_back(*itr, comment_api_object());
+            }
+        };
+
+        if (ReverseSort) {
+            for (; itr != idx.begin() && result.size() < query.limit; --itr) {
+                handle();
+            }
```

Comment:
This case doesn't work.
idx.end() - will return pointer to invalid data.

You should use `std::make_reverse_iterator(itr)`